File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/java/field_assign_to_deref_in_function.java
class Foo {
    String field;
}
class A {
    void func(Foo fo111, Foo fo222)
    {
        fo111.field = fo222.field;
    }

    int main()
    {
        Foo f11 = new Foo();
        Foo f22 = new Foo();
        f22.field = null;
        func(f11, f22);
        return f11.field.length();  // should be detect here
    }
}

File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/field_assign_to_deref_in_function.cpp
class Foo {
public:
    int *field;
};

void func(Foo *fo111, Foo *fo222) {
    fo111->field = fo222->field;
}

int main() {
    Foo f11;
    Foo f22;
    f22.field = 0;
    func(&f11, &f22);
    return *f11.field; // should be detect here
}
